### DISCUSSION TITLE
Using 'broadcast-safe' colors

### DISCUSSION THREAD

**Discussion by JoeD4321** [ImageMagick/ImageMagick#7588] [General] [answered]
Hello all  -  I was wondering if magick had any functionality to create images with broadcast-safe colors or to verify that a color palette is broadcast safe. A color palette is 'broadcast safe' if none of the RGB colors have any red, green or blue values below 0x10 or above 0xEB.  I am worried that if I use imagemagick to perform an operation on a broadcast-safe image, how do II know it is still broadcast safe?

My research has not yielded any clues.  Thanks in advance

**Comment by snibgo** [comment, marked as answer]
0x10 is decimal 16. 0xEB is decimal 235. I assume these value are out of a nominal 0 to 255. Below, values are normalised to the range 0 to 1, so we compare values to 10/255 and 235/255.

We can find the minimum pixel value. If that value is less than 16, the image is not "broadcast-safe". Likewise if the maximum value is more than 235. The following returns 0 if the image is not safe, or 1 if it is.
```
magick toes.png -format "%[fx:minima<10/255?0:maxima>235/255?0:1]" info:
0

magick -size 100x100 xc:gray50 -format "%[fx:minima<10/255?0:maxima>235/255?0:1]" info:
1
```
My image `toes.png` is not "broadcast-safe". An image that is entirely "gray50" is safe.

We can clamp all pixel values to be within the range like this:
```
magick toes.png -fx "lo=%[fx:10/255];u<lo?lo:u" -fx "hi=235/255;u>hi?hi:u" x.png
```
Check that x.png is safe:
```
D:\temp>%IMG7%magick x.png -format "%[fx:minima<10/255?0:maxima>235/255?0:1]" info:
1
```
As expected, x.png is safe.

Note that clamping is crude. It will change hues.

**Comment by snibgo** [comment]
Yes, thanks for the correction.
